### PR TITLE
AG-7585 Treemap layout improvements

### PR DESCRIPTION
**Pull request by alexanderby**
- Use Squarified Treemap algorithm with 1 ratio.
- Sort nodes by size.
- Crop overflown labels.
- Use proper Datum in Treemap tooltip.

### CODE REVIEW

**Review by Rajusheik — APPROVED** [review]

File: charts-packages/ag-charts-community/src/chart/series/hierarchy/treemapSeries.ts [hunk 1]
```diff
@@ -196,78 +190,197 @@ export class TreemapSeries extends HierarchySeries<TreemapNodeDatum> {
...
+            right: nodePadding,
+            bottom: nodePadding,
+            left: nodePadding,
         };
     }
 
+    /**
+     * Squarified Treemap algorithm
+     */
+    private squarify(
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/chart/series/hierarchy/treemapSeries.ts:228]: Could you give reference to the source for the algorithm here?

File: charts-packages/ag-charts-community/src/chart/series/hierarchy/treemapSeries.ts [hunk 2]
```diff
@@ -196,78 +190,197 @@ export class TreemapSeries extends HierarchySeries<TreemapNodeDatum> {
...
+                    partition.width,
+                    (partition.height * children[i].value) / partitionSum
+                );
+                this.squarify(children[i], childBox, outputNodesBoxes);
+                startY += childBox.height;
+            }
+        }
+
+        return outputNodesBoxes;
+    }
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/chart/series/hierarchy/treemapSeries.ts:341]: Not sure if it's worth it, but could we simplify and/or break this method up a bit for readability? Happy to do this in a follow-up PR so we can put a stake in the ground first.

**Review by alantreadway — COMMENTED** [review]
Nice work!

Just in terms of admin, I assume there are going to be further changes incoming in the scope of AG-7585?

Could you make sure to capture any gaps or changes in the options contract as part of that JIRA, and/or break into distinct JIRA(s) for follow-up and prioritisation perhaps?

File: charts-packages/ag-charts-community/src/chart/series/hierarchy/treemapSeries.ts (hunk 1 above)
Comment (alexanderby) [reply charts-packages/ag-charts-community/src/chart/series/hierarchy/treemapSeries.ts:229]: Added a link to the doc

File: charts-packages/ag-charts-community/src/chart/series/hierarchy/treemapSeries.ts (hunk 2 above)
Comment (alexanderby) [reply charts-packages/ag-charts-community/src/chart/series/hierarchy/treemapSeries.ts]: Done

**Pull request merged**